src/txn/txn_recover.c

```diff
@@ -974,8 +1005,19 @@ __wt_txn_recover(WT_SESSION_IMPL *session, const char *cfg[])
          */
         WT_ERR(session->iface.checkpoint(&session->iface, "force=1"));
 
-    /* Initialize the connection's base write generation after rollback to stable. */
-    WT_ERR(__wt_metadata_init_base_write_gen(session));
+    /*
+     * To support rollback to stable operation to fix the possible inconsistent checkpoint data on
+     * disk that is possible to be written when eviction runs in parallel to the checkpoint with a
+     * newer transaction snapshot than the checkpoint thread. To do that the transaction ids that
```

Comment:
That first sentence is very long and very confusing. Is it needed? I don't think this code should discuss how another subsystem works because then it may be out of date when that code is updated and no one knows to rewrite a comment here. I think this paragraph could be removed.

Reply:
Removed.